plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -381,8 +381,7 @@ private static LocalTime toLocalTime(Time sqlTime)
     {
         // Time.toLocalTime() does not preserve second fraction
         return sqlTime.toLocalTime()
-                // TODO is the conversion correct if sqlTime.getTime() < 0?
-                .withNano(toIntExact(MILLISECONDS.toNanos(sqlTime.getTime() % 1000)));
+                .withNano(toIntExact(MILLISECONDS.toNanos(Math.floorMod(sqlTime.getTime(), 1000L))));
```

Comment:
the commit message

> Support negative SQL time ...


suggests we're dealing with values before `00:00:00` (which would be incorrect, such TIME values do not exist). Please reword to something like

```
Fix TIME read function when JVM zone east of Greenwich
```